### PR TITLE
Use PSR-11 Container Interface

### PR DESCRIPTION
**Pull request by geerteltink**
container-interop can't be removed completely...  `ZendViewRendererFactoryTest` has tests for `Zend\ServiceManager\AbstractPluginManager`, which is not updated yet.

### CODE REVIEW

**Comment by michaelmoussa** [comment]
@xtreamwayz this looks good (so far) but I suggest waiting on merging this until `container-interop` _can_ be removed completely. The partial changes are already a BC break, and removing `container-interop` entirely later would be another BC break. Might as well do it all in one shot.

There are quite a number of references to it in `zendframework/zend-servicemanager` which likely tie in to other ZF components, but as long as it gets a major version bump, we shouldn't need to update the other components right away.

**Comment by michaelmoussa** [comment]
I went ahead and submitted a PR to both [`zendframework/zend-servicemanager`](https://github.com/zendframework/zend-servicemanager/pull/187) and [`zendframework/zend-eventmanager`](https://github.com/zendframework/zend-eventmanager/pull/52) for interop -> PSR-11. Maybe we can end up removing interop from this PR entirely like you were able to do for the other two renderer repositories.

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     },
     "require": {
         "php": "^5.6 || ^7.0",
-        "container-interop/container-interop": "^1.2",
+        "container-interop/container-interop": "^1.2.0",
```
Comment (michalbundyra) [inline composer.json:22]: We don't need above line. We are not using this library here anymore.
We need of course `conflict` entry, as it is now below.

File: src/Exception/MissingHelperException.php [hunk 2]
```diff
@@ -8,10 +8,10 @@
 namespace Zend\Expressive\ZendView\Exception;
 
 use DomainException;
-use Interop\Container\Exception\ContainerException;
+use Psr\Container\ContainerExceptionInterface;
 
 class MissingHelperException extends DomainException implements
-    ContainerException,
+    ContainerExceptionInterface,
```
Comment (michalbundyra) [inline src/Exception/MissingHelperException.php:14]: Isn't it here BC Break? If somebody catch exception on `ContainerException` it will stop work.

**Review by michalbundyra — COMMENTED** [review]
I think it should be marked as BC Break, because we change type of exception. Maybe type hint in factories is not a big issue, don't think so that somebody extends these factories and override `__invoke` method. But anyway... it's still possible.

**Review by mwillbanks — APPROVED** [review]

**Comment by mwillbanks** [comment]
At this point this is starting to look really good.  I do think we need to support PSR-11 properly, there are some other comments that *might* be necessary to look at prior.  My approval is strictly on the code and the implements.

File: src/Exception/MissingHelperException.php (hunk 2 above)
Comment (weierophinney) [reply src/Exception/MissingHelperException.php:14]: If they were catching `Interop\Container\Exception\ContainerException`, however, and the container in use _implements PSR-11 only_, then there's breakage.

**Comment by weierophinney** [comment]
Thanks, @xtreamwayz!

**Pull request merged**